front/app/containers/IdeasEditPage/IdeasEditForm.tsx

```diff
@@ -233,13 +232,13 @@ const IdeasEditForm = ({ ideaId }: Props) => {
     const isImageNew =
       idea_images_attributes !== initialFormData.idea_images_attributes;
 
-    // Delete a remote image only on submission
-    // TODO: Fix this the next time the file is edited.
-    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-    if (isImageNew && initialFormData?.idea_images_attributes[0]?.id) {
-      deleteIdeaImage({
-        ideaId,
-        imageId: initialFormData.idea_images_attributes[0].id,
+    // Delete remote images only on submission
+    if (isImageNew && initialFormData.idea_images_attributes.length > 0) {
+      initialFormData.idea_images_attributes.forEach((image) => {
+        deleteIdeaImage({
+          ideaId,
+          imageId: image.id,
+        });
```

Comment:
In some cases ideas might accidentally have more than 1 image in `idea_images_attributes` (E.g. From past bugs), so I've changed this to just make sure we delete any old images from the idea if we're adding a new one :)